.github/workflows/build.yml

```diff
@@ -32,15 +41,32 @@ jobs:
           const latestDevCommit = "${{ steps.latest_commit.outputs.commit }}"
           const { owner, repo } = context.repo
           let { data: release } = await github.rest.repos.getLatestRelease({ owner, repo })
-          const latestReleaseCommit = release.body.split('@')[1]
+          const firstLine = release.body.split('\n')[0]
+          const latestReleaseCommit = firstLine.split('@')[1]
           console.log(`Latest release commit: ${latestReleaseCommit}`)
           console.log(`Latest ruby commit: ${latestDevCommit}`)
-          if (latestReleaseCommit === latestDevCommit) {
-            return 'false'
-          } else {
-            return 'true'
-          }
-        result-encoding: string
+          core.setOutput('should_build', latestReleaseCommit !== latestDevCommit)
+          core.setOutput('previous_release', release.tag_name)
+    - name: Compute build and reuse matrix
+      uses: actions/github-script@v7
+      id: matrix
+      with:
+        script: |
+          const osList = ['ubuntu-20.04', 'ubuntu-22.04', 'ubuntu-24.04', 'ubuntu-22.04-arm', 'ubuntu-24.04-arm', 'macos-13', 'macos-14']
+          const skipSlow = "${{ github.event_name == 'workflow_dispatch' && github.event.inputs.skip_slow == 'true' }}"
+          const buildMatrix = JSON.stringify({
+            os: osList,
+            name: skipSlow === 'false' ? ['head', 'debug'] : ['head'],
+            ...(skipSlow === 'false' && { include: [{ os: 'ubuntu-24.04', name: 'asan' }] })
+          })
+          core.setOutput('build_matrix', buildMatrix)
+          const reuseMatrix = JSON.stringify({
```

Comment:
Can we make this one empty if `skipSlow===false`?

Reply:
This is where the 

> I think the final result is quite simple/maintainable, although I'll admit it took quite a few iterations to get to this point.

I mentioned above shows up lol.

TL;DR: Tried it on an earlier attempt, GHA doesn't like empty matrix. (Skips all jobs that depend on it)

We could have a smaller matrix, but that seems a bit of an extra workaround as these steps will run quickly anyway.

Reply:
Ah that's unfortunate.
I'm mostly concerned with the confusion in like https://github.com/DataDog/ruby-dev-builder/actions/runs/13326349375 where it looks like it resued builds because of the presence of that reuse matrix but actually it didn't.

From your comment `# We can't skip the whole job as publish depends on it, but we skip the uploading` I guess using `if:` on the job doesn't work either.

I think the next best thing then would be to make it a matrix with a single entry, and for that entry use a name like `noop` or so.

Reply:
Could you also add a source comment that an empty matrix prevents the publish job to run?

Reply:
Added in 9efa7556fde1f8bf232282fdb41f06be7cb25094 , example build https://github.com/DataDog/ruby-dev-builder/actions/runs/13370294484